synapse/events/snapshot.py

```diff
@@ -13,56 +13,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import abc
+
 from frozendict import frozendict
 
 from twisted.internet import defer
 
 
-class EventContext(object):
+class StatelessContext(object):
     """
     Attributes:
-        current_state_ids (dict[(str, str), str]):
-            The current state map including the current event.
-            (type, state_key) -> event_id
-
-        prev_state_ids (dict[(str, str), str]):
-            The current state map excluding the current event.
-            (type, state_key) -> event_id
-
         state_group (int|None): state group id, if the state has been stored
             as a state group. This is usually only None if e.g. the event is
             an outlier.
         rejected (bool|str): A rejection reason if the event was rejected, else
             False
 
-        push_actions (list[(str, list[object])]): list of (user_id, actions)
-            tuples
-
         prev_group (int): Previously persisted state group. ``None`` for an
             outlier.
         delta_ids (dict[(str, str), str]): Delta from ``prev_group``.
             (type, state_key) -> event_id. ``None`` for an outlier.
 
         prev_state_events (?): XXX: is this ever set to anything other than
             the empty list?
+
+        current_state_ids (dict[(str, str), str]|None):
+            The current state map including the current event.
```

Comment:
could we record the circumstances under which we expect it to be set, and which None? (likewise prev_state_ids)